demos/mandelbrot/main.cpp

```diff
@@ -0,0 +1,149 @@
+/***************************************************************************
+ *
+ *  Copyright (C) 2017 Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  Codeplay's ComputeCpp SDK
+ *
+ *  main.cpp
+ *
+ *  Description:
+ *    Application description for Mandelbrot demo.
+ *
+ **************************************************************************/
+
+#include <iostream>
+
+#include <cinder/CameraUi.h>
+#include <cinder/TriMesh.h>
+#include <cinder/app/App.h>
+#include <cinder/app/RendererGl.h>
+#include <cinder/gl/gl.h>
+
+#include <CL/sycl.hpp>
+namespace sycl = cl::sycl;
+
+#include <codeplay_demo.hpp>
+
+#include "mandel.hpp"
+
+
+constexpr size_t WIDTH = 800;
+constexpr size_t HEIGHT = 600;
+
+class MandelbrotApp
+#ifdef CODEPLAY_DRAW_LOGO
+    : public CodeplayDemoApp
+#else
+    : public ci::app::App
+#endif
+{
+  // Use doubles for more zoom
+  MandelbrotCalculator<double> m_calc;
```

Comment:
note that doubles will cause a major drop in performance in many accelerators , can you make doubles optional? maybe parametrize the execution?